git/pre-commit

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+set -e
+
+# This file will exist in $project_root/.git/hooks/, so we need to move back up
+# to $project_root. This apparent mess needs to happen because macOS doesn't
+# have `readlink -f`.
+declare src="$0"
+declare dir=''
+
+while [ -h "$src" ]; do
+  dir="$(cd -P "$( dirname "$src")" && pwd)"
+  src="$(readlink "$src")"
+  [[ "$src" != /* ]] && src="$dir/$src"
+done
+
+cd -P "$(dirname "$src")/../"
+
+##### This is the actual pre-commit hook. #####
+
+if [ -z "$VIRTUAL_ENV" ]; then
+    # If the developer has a virtual environment in the canonical location, then
+    # this will attempt to use that. Otherwise, assume the necessary things are
+    # installed globally.
+    declare -r venv='./venv/bin/activate'
+    if [ -f "$venv" ]; then
+      source "$venv"
+    fi
+fi
+
+pwd
```

Comment:
Nit: remove the `pwd`? Looks like that's left over from testing. New version works for me as a symlink! (Also tested removing the `pwd` and committing again—still works.)

Reply:
Done!